### PR TITLE
Back button for overlays and modals / alerts

### PR DESCRIPTION
**Pull request by parnas**
#811

### CODE REVIEW

File: app/router.js [hunk 1]
```diff
@@ -767,7 +775,56 @@ define(function (require, exports, module) {
...
+        if(subroutes[previousSubrouteId].type == 'overlay') {
+          this.removeOverlayPage(subroute.view);
+        }
+      }
+
+      this.routeHistory.subrouteId = newSubrouteId;
+    },
+
     initialize: function (options) {
+      window.onpopstate = function (event) {
```
Comment (rijkvanzanten) [inline app/router.js]: Could we change this to `window.addEventListener('popstate' ... )` to prevent any overriding issues in the future?

**Review by rijkvanzanten — APPROVED** [review]
lgtm except for the one little nitpicky thingy ✌🏻

**Review by rijkvanzanten — APPROVED** [review]

**Comment by WellingGuzman** [comment]
Awesome @parnas, it's working great!

**Comment by danibaena** [comment]
Hi everybody, these changes are related to a bug where you can't delete a column in a table from the table view (at route yourhost/settings/tables/*)

I've successfully undone the change in line 214 in router.js, replacing `this.openViewInModal(view)` with its previous version `containerView.show(view)`, and now it allows to delete the columns. Not sure if history is broken having restored this change, maybe it's worth taking a deeper look, but I just needed the delete column feature back.

**Comment by parnas** [comment]
Confirmed. Thank you for the report @danibaena I'm looking into that.

**Pull request merged**